src/signaling/backend_configuration.go

```diff
@@ -146,26 +146,28 @@ func (b *BackendConfiguration) UpsertHost(host string, backends []*Backend) {
 	b.backends[host] = append(b.backends[host], backends...)
 }
 
-func getConfiguredBackendIDs(config *goconf.ConfigFile) (ids map[string]bool) {
-	ids = make(map[string]bool)
+func getConfiguredBackendIDs(backendIds string) (ids []string) {
+	seen := make(map[string]bool)
 
-	if backendIds, _ := config.GetString("backend", "backends"); backendIds != "" {
-		for _, id := range strings.Split(backendIds, ",") {
-			id = strings.TrimSpace(id)
-			if id == "" {
-				continue
-			}
+	for _, id := range strings.Split(backendIds, ",") {
+		id = strings.TrimSpace(id)
+		if id == "" {
+			continue
+		}
 
-			ids[id] = true
+		if seen[id] {
+			continue
 		}
+		ids = append(ids, id)
+		seen[id] = true
 	}
 
 	return ids
```

Comment:
Just a remark: This guarantees the ids to be in config order. Could it be useful to also sort the ids?

Reply:
I would rather have the backends being processed in the order they are configured in the file.